lib/iris/analysis/maths.py

```diff
@@ -1085,11 +1112,27 @@ def wrap_data_func(*args, **kwargs):
 
             return self.data_func(*args, **kwargs_combined)
 
-        if self.nin == 2:
+        if self.nin == 1:
+            if other is not None:
+                dmsg = (
+                    "ignoring surplus 'other' argument to IFunc.__call__, "
+                    f"provided data_func {self._data_func_name!r} only requires "
+                    "1 input"
+                )
+                logger.debug(dmsg)
```

Comment:
Create a `logging` debug message rather than raising a `ValueError`, which now aligns the `IFunc.__call__` behaviour for a `data_func.nin == 1` to that of `apply_ufunc` with a `ufunc.nin == 1`.
This will have an appropriate `whatsnew` entry for the community.